File: bot/app/model/report.py
from dataclasses import dataclass, field
from datetime import datetime
from typing import List


@dataclass
class Report:
    """Модель жалобы на объявление"""
    id: int
    offer_id: int
    reporter_id: int
    reason: str
    created_at: datetime


@dataclass
class ReportCreate:
    """Данные для создания жалобы"""
    offer_id: int
    reporter_id: int
    reason: str

File: api/internal/model/report.go
package model

import "time"

type Report struct {
	Id         int64     `json:"id"`
	OfferId    int64     `json:"offer_id"`
	ReporterId int64     `json:"reporter_id"`
	Reason     string    `json:"reason"`
	Status     string    `json:"status"`
	CreatedAt  time.Time `json:"created_at"`
}

type ReportCreate struct {
	OfferId    int64  `json:"offer_id" binding:"required"`
	ReporterId int64  `json:"reporter_id" binding:"required"`
	Reason     string `json:"reason" binding:"required"`
}
